### DISCUSSION TITLE
Add request object to context at hooks

### DISCUSSION THREAD

**Discussion by GalStan** [directus/directus#11596] [Feature Requests]
I need to implement "login as another user" functionality and I want to use custom headers and/or cookies, so it would be great to have such a possibility inside the filter hook.

**Comment by rijkvanzanten** [comment]
So if I'm understanding you correctly, you're looking for the current request object inside a filter hook? 🤔

**Reply by emcdaniel-sungage** [reply]
Hi @rijkvanzanten, yes that's essentially what we're looking for (I work with @GalStan).

**Reply by rijkvanzanten** [reply]
Copy that! Glad to report that we currently have a new filter hook in the works that would allow for exactly this, so keep an eye on the release notes 🙂  (I'll post a link to the PR here when it's ready)

**Reply by emcdaniel-sungage** [reply]
Great, thanks @rijkvanzanten

**Reply by rijkvanzanten** [reply]
Promised link: https://github.com/directus/directus/pull/11942 to PR

**Reply by GalStan** [reply]
Thanks a lot! Looks good

**Reply by leandroromano** [reply]
Hi! Is it possible to add the current request object inside a filter hook of `read` event ?

**Reply by tan-tan-kanarek** [reply]
PR: https://github.com/directus/directus/pull/17173

**Comment by rijkvanzanten** [comment]
Heya! Thank you for taking the time to submit this request! 

It has been over **90 days**, and this discussion has not received at least **15 votes** from the community. This means that we don't feel like there's enough community interest to warrant further R&D into this topic at this time. :ice_cube:

This request will now be closed to keep our discussions tidy. Please reach out if you have any questions!

_For more information, see our [Feature Request Process](https://docs.directus.io/contributing/feature-request-process.html)._
